packages/@lwc/engine/src/framework/local-registry.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { appendVM, removeVM, getAssociatedVMIfPresent, renderVM } from './vm';
+import { isUndefined } from '@lwc/shared';
+import { startGlobalMeasure, GlobalMeasurementPhase, endGlobalMeasure } from './performance-timing';
+
+const globalRegisteredNames: Set<string> = new Set();
+
+/**
+ * LWC Upgradable Element reference to an element that was created
+ * via the scoped registry mechanism, and that is ready to be
+ * upgrade by calling createVM operation on it.
+ */
+class LWCUpgradableElement extends HTMLElement {
+    connectedCallback() {
+        const vm = getAssociatedVMIfPresent(this);
+        if (!isUndefined(vm)) {
+            startGlobalMeasure(GlobalMeasurementPhase.HYDRATE, vm);
```

Comment:
This shouldn't be here, because native custom elements will not be marked like other components. 
I simplified this all this code in #1751. Is it possible to rebase this PR on top of master, since a lot of the engine internals changed?